aiida/cmdline/commands/cmd_setup.py

```diff
@@ -54,6 +54,9 @@ def setup(
     from aiida import orm
     from aiida.manage.configuration import get_config
 
+    # store default user settings so user does not have to re-enter them
+    config = _store_default_user_settings(non_interactive, email, first_name, last_name, institution)
```

Comment:
The config is being fetched in this function, but then it is retrieved once again on line 83. I would propose to get the config once in this function, and pass it as the first argument to `_store_default_user_settings`.

Better yet, the context is passed in the `ctx` argument passed to the click command. I would simply use this and get rid of the `get_config` altogether. So simply call
```python
_store_default_user_settings(ctx.obj.config, email, first_name, last_name, institution)
```
and also use `ctx.obj.config` elsewhere in the command.



Also, please add a test because this is currently failing. You are passing in `non_interactive` but the function doesn't accept it.

Reply:
Thanks for this suggestions, I didn't know the config was automatically shipped in the context.

The function is tested by the setup/quicksetup tests.
If desired, I could still add a test to check whether the info is stored even if quicksetup fails

Reply:
>  If desired, I could still add a test to check whether the info is stored even if quicksetup fails

Yes, this is what I meant, would be great if you could add it. It should be relatively straightforward to test it. Just pass non-sensical values for db-connection using the `empty_config` fixture to guarantee an empty config, and then check that the relevant autofill keys in the `Config` are set.

Reply:
Thanks @ltalirz . I had some suggestions to improve the test, but it affected lines not covered by the PR, so instead I added a commit, hope you don't mind. Let me know if you are happy with the change, then I will merge this PR.